sdk/core/Azure.Core/src/DynamicData/MutableJsonDocument.cs

```diff
@@ -13,6 +14,9 @@ namespace Azure.Core.Json
     /// <summary>
     /// A mutable representation of a JSON value.
     /// </summary>
+#if !NET5_0
+    [RequiresUnreferencedCode("This class utilizes reflection-based JSON serialization and deserialization which is not compatible with trimming.")]
+#endif
```

Comment:
```suggestion
    [RequiresUnreferencedCode("This class utilizes reflection-based JSON serialization and deserialization which is not compatible with trimming.")]
```

These `#if`s shouldn't be necessary.

Reply:
Oh, it is necessary because `net5.0` doesn't have `AttributeTargets.Class` on the attribute.

Can you add a comment here? like we do in 

https://github.com/Azure/azure-sdk-for-net/blob/9f5b3bd7051bf09124919da9897864af34337901/sdk/core/Azure.Core/src/Serialization/ModelJsonConverter.cs#L15-L18